pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -430,20 +465,28 @@ func (w *wlReconciler) nominateAndSynchronizeWorkers(ctx context.Context, group
 	log.V(3).Info("Nominate and Synchronize Worker Clusters")
 	var nominatedWorkers []string
 
-	if w.dispatcherName == config.MultiKueueDispatcherModeAllAtOnce {
-		for workerName := range group.remotes {
-			nominatedWorkers = append(nominatedWorkers, workerName)
-		}
-		if group.local.Status.ClusterName == nil && !equality.Semantic.DeepEqual(group.local.Status.NominatedClusterNames, nominatedWorkers) {
-			group.local.Status.NominatedClusterNames = nominatedWorkers
-			if err := workload.ApplyAdmissionStatus(ctx, w.client, group.local, true, w.clock); err != nil {
-				log.V(2).Error(err, "Failed to patch nominated clusters", "workload", klog.KObj(group.local))
-				return reconcile.Result{}, err
+	// For elastic workloads, retrieve the remote cluster where the original workload was scheduled.
+	// For now, new workload slices will continue to be assigned to the same cluster.
+	// In the future, we may introduce more nuanced remote workload propagation policies,
+	// supporting preferred or required placement constraints.
+	if clusterName := workload.ClusterName(group.local); group.IsElasticWorkload() && clusterName != "" {
+		nominatedWorkers = []string{clusterName}
+	} else {
+		if w.dispatcherName == config.MultiKueueDispatcherModeAllAtOnce {
+			for workerName := range group.remotes {
+				nominatedWorkers = append(nominatedWorkers, workerName)
+			}
+			if group.local.Status.ClusterName == nil && !equality.Semantic.DeepEqual(group.local.Status.NominatedClusterNames, nominatedWorkers) {
+				group.local.Status.NominatedClusterNames = nominatedWorkers
+				if err := workload.ApplyAdmissionStatus(ctx, w.client, group.local, true, w.clock); err != nil {
+					log.V(2).Error(err, "Failed to patch nominated clusters", "workload", klog.KObj(group.local))
+					return reconcile.Result{}, err
+				}
```

Comment:
This looks racy, as we update the NominatedClusterNames for the replacement workload which is already created. So, before we act it may get assigned to another cluster.

I think the proper solution would be to set the clusterName up front when creating the replacement Workload. Then, basically skip the nomination phase. Wdyt?

Reply:
wonder if I messed it up during rebase. 
I will take another (closer) look.

Reply:
I am sorry, responded incorrectly about "diff" here, instead of next comment. I moved my message(s) to[ the correct thread.](https://github.com/kubernetes-sigs/kueue/pull/6445#pullrequestreview-3241375600)